src/controllers/dataTierApplicationWebviewController.ts

```diff
@@ -0,0 +1,992 @@
+                // (e.g., user cancelled password prompt)
+                ownerUri = this.connectionManager.getUriForConnection(profile);
+                const connectionInfo = ownerUri
+                    ? this.connectionManager.activeConnections[ownerUri]
+                    : undefined;
+                const errorMessage = connectionInfo?.errorMessage || "Failed to connect to server";
+                return {
+                    ownerUri: "",
+                    isConnected: false,
+                    errorMessage,
+                };
+            }
+        } catch (error) {
+            this.logger.error(`Failed to connect to server: ${error}`);
+            const errorMessage = error instanceof Error ? error.message : String(error);
+            return {
+                ownerUri: "",
+                isConnected: false,
+                errorMessage: `Connection failed: ${errorMessage}`,
+            };
+        }
+    }
+
+    /**
+     * Builds a display name for a connection profile
+     */
+    private buildConnectionDisplayName(profile: IConnectionProfile): string {
+        let displayName = profile.profileName || profile.server;
+        if (profile.database) {
+            displayName += ` (${profile.database})`;
+        }
+        if (profile.user) {
+            displayName += ` - ${profile.user}`;
+        }
+        return displayName;
+    }
+
+    /**
+     * Gets a string representation of the authentication type
+     */
+    private getAuthenticationTypeString(authType: number | string | undefined): string {
+        switch (authType) {
+            case 1:
+                return "Integrated";
+            case 2:
+                return "SQL Login";
+            case 3:
+                return "Azure MFA";
+            default:
+                return "Unknown";
+        }
+    }
+
+    /**
+     * Validates a database name
+     */
+    private async validateDatabaseName(
+        databaseName: string,
+        ownerUri: string,
+        shouldNotExist: boolean,
+        operationType?: DataTierOperationType,
+    ): Promise<{ isValid: boolean; errorMessage?: string }> {
+        if (!databaseName || databaseName.trim() === "") {
+            return {
+                isValid: false,
+                errorMessage: LocConstants.DataTierApplication.DatabaseNameRequired,
+            };
+        }
+
+        // Check for invalid characters
+        const invalidChars = /[<>*?"/\\|]/;
+        if (invalidChars.test(databaseName)) {
+            return {
+                isValid: false,
+                errorMessage: LocConstants.DataTierApplication.InvalidDatabaseName,
+            };
+        }
+
+        // Check length (SQL Server max identifier length is 128)
+        if (databaseName.length > 128) {
+            return {
+                isValid: false,
+                errorMessage: LocConstants.DataTierApplication.DatabaseNameTooLong,
+            };
+        }
```

Comment:
These seem like useful utility that we can use elsewhere; can you extract them to helper functions in `src\models\utils.ts`?